### PR TITLE
Fixed some qt Issues

### LINKED ISSUE #48: On-map worked tile chooser doesn't highlight the correct tiles
It can highlight too little, or too much. In the picture below tiles on the left are missing to create a complete hexagon, but too many tiles are highlighted (the working radius is 1 and the vision radius is 2, alio ruleset and tileset):
image

### PR DESCRIPTION
**Pull request by mir3x**
It should be close #48 not 49 in b45f0bc

### CODE REVIEW

File: client/tilespec.cpp [hunk 1]
```diff
@@ -5224,13 +5224,14 @@ static int fill_grid_sprite_array(
...
       }
-        if (tile) {
-          { // Draw city grid for main citymap
-            xcity = find_city_near_tile(tile);
-            if (xcity && xcity->client.city_opened) {
-              ADD_SPRITE_SIMPLE(t->sprites.grid.selected[pedge->type]);
-            }
+      if (tile) {
+        { // Draw city grid for main citymap
+          xcity = find_city_near_tile(tile);
```
Comment (lmoureaux) [inline client/tilespec.cpp:5229]: I think it might still fail because `find_city_near_tile` always returns the closest city?

Comment (mir3x) [reply client/tilespec.cpp:5229]: comment says that:
   Find the "best" city/settlers to associate with the selected tile.
     a.  If a visible city is working the tile, return that city.
     b.  If another player's city is working the tile, return NULL.
     c.  If any selected cities are within range, return the closest one.
     d.  If any cities are within range, return the closest one.
     e.  If any active (with color) settler could work it if they founded a
         city, choose the closest one (only if punit != NULL).
     f.  If any settler could work it if they founded a city, choose the
         closest one (only if punit != NULL).
     g.  If nobody can work it, return NULL.

Comment (mir3x) [reply client/tilespec.cpp:5229]: I tried with some cities near and was working as expected but maybe I was lucky

Comment (lmoureaux) [reply client/tilespec.cpp:5229]: ok let's merge and reopen the issue if it's still there

Comment (mir3x) [reply client/tilespec.cpp:5229]: or maybe I jsut iterate to be sure

**Pull request merged**